packages/gas-fee-controller/src/GasFeeController.ts

```diff
@@ -240,12 +240,10 @@ type AllowedActions =
   | NetworkControllerGetNetworkClientByIdAction
   | NetworkControllerGetEIP1559CompatibilityAction;
 
-type GasFeeMessenger = RestrictedMessenger<
+export type GasFeeMessenger = Messenger<
   typeof name,
   GasFeeControllerActions | AllowedActions,
-  GasFeeControllerEvents | NetworkControllerNetworkDidChangeEvent,
-  AllowedActions['type'],
-  NetworkControllerNetworkDidChangeEvent['type']
+  GasFeeControllerEvents | NetworkControllerNetworkDidChangeEvent
```

Comment:
Nit: For consistency with other controllers should we extract `NetworkControllerNetworkDidChangeEvent` to an `AllowedEvents` type?